src/test/java/be/syntra/devshop/DevshopBack/controllers/ProductControllerTest.java

```diff
@@ -56,99 +57,55 @@
     @MockBean
     private CategoryServiceImpl categoryService;
 
+    @MockBean
+    private SearchService searchService;
+
+    @MockBean
+    private SearchModelMapperUtility searchModelMapperUtility;
+
     @Mock
     private SecurityUserFactory securityUserFactory;
 
     @MockBean
     private SecurityUserService securityUserService;
 
-    @Autowired
+    @MockBean
     private ProductMapperUtility productMapperUtility;
 
     @Autowired
     private CategoryMapperUtility categoryMapperUtility;
 
     @Test
     @WithMockUser
-    void testRetrieveAllNonArchivedProductsEndpoint() throws Exception {
+    void saveProductTest() throws Exception {
```

Comment:
saveProductTest() and canUpdateProductTest can be converted to 
@ParameterizedTest
    @ValueSource(strings = {"/products", "/products/update"})
Keep the body of the updateTest. Remove the @Test, place the annotations I provided above. Remove the saveTest. Add 'String url' as param of the test. Use 'url' instead of hardcoded String-value as urlTemplate

Reply:
changed, nice!